### PR TITLE
Adding palo cortex xdr intake

### PR DESCRIPTION
**Pull request by m-242**

### CODE REVIEW

**Comment by gbossert** [comment]
Awesome :clap: Thank you very much for your contribution.

**Comment by m-242** [comment]
Les tests automatisés passent localement, donc je suppose qu'ils passeront également ici.

Ceci dit, il n'y a pour le moment qu'un seul test unitaire de l'intake, en rajouter quelques uns me semble nécéssaire, je ne sais juste pas encore lesquels

**Comment by m-242** [comment]
Added another case of test, I don't see much else to add, but suggestions are welcome

File: Palo Alto Networks/paloalto-cortex/_meta/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+uuid: 00000000-0000-0000-0000-000000000000
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/manifest.yml:1]: You can provide any non-existing (in this repository) uuid
```suggestion
uuid: 2652d6b3-242b-4561-9b34-4896f75d3770
```

File: Palo Alto Networks/paloalto-cortex/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        output_field: message
+  - name: setter
+    description: ''
+    filter: ''
+    external: null
+stages:
+  setter:
+    actions:
+      - set:
+          cortexxdr.shost: '{{parsed_event.message.shost}}'
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/ingest/parser.yml:19]: Please use [ECS fields](https://www.elastic.co/guide/en/ecs/8.4/ecs-event.html) as the base of your taxonomy.
For example, here, [`host.name`](https://www.elastic.co/guide/en/ecs/8.4/ecs-host.html#field-host-name) is more suitable.
```suggestion
          host.name: '{{parsed_event.message.shost}}'
```

File: Palo Alto Networks/paloalto-cortex/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          cortexxdr.suser: '{{parsed_event.message.suser}}'
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/ingest/parser.yml:20]: According to your tests, `suser` contains the username and the user's domain (e.g: `domain\\user`).
I suggest extracting this two information:
```suggestion
          user.name: '{{parsed_event.message.suser.split("\\"}[1]}'
           user.domain: '{{parsed_event.message.suser.split("\\"}[0]}'
```

File: Palo Alto Networks/paloalto-cortex/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          cortexxdr.action: '{{parsed_event.message.act|lower}}'
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/ingest/parser.yml:21]: [`event.action`](https://www.elastic.co/guide/en/ecs/8.4/ecs-event.html#field-event-action) is more suitable:
```suggestion
          event.action: '{{parsed_event.message.act|lower}}'
```

File: Palo Alto Networks/paloalto-cortex/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          cortexxdr.category: '{{parsed_event.message.cat|upper}}'
+          cortexxdr.event_subtype: '{{parsed_event.message.subtype}}'
+          cortexxdr.severity_human: '{{parsed_event.message.Severity}}'
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/ingest/parser.yml:24]: [`event.severity`](https://www.elastic.co/guide/en/ecs/8.4/ecs-event.html#field-event-severity) seems more suitable:
```suggestion
          event.severity: '{{parsed_event.message.Severity}}'
```

File: Palo Alto Networks/paloalto-cortex/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          cortexxdr.original_message: '{{parsed_event.message.msg}}'
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/ingest/parser.yml:25]: Original message is already present in the resulting event (in the message field).
```suggestion
```

File: Palo Alto Networks/paloalto-cortex/ingest/parser.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        filter: ''
+        name: set
+      - fallback: UNPARSED
+        dictionary:
+          '0': UNKNOWN
+          '6': LOW
+          '8': MEDIUM
+          '9': HIGH
+        mapping:
+          cortexxdr.severity_human: cortexxdr.severity_human
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/ingest/parser.yml:35]: According to the suggestion above:
```suggestion
          event.severity: cortexxdr.severity_human
```

File: Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
+[
+  {
+    "value": "{cortexxdr.category}: An event of {cortexxdr.severity_human} severity was {cortexxdr.action} on endpoint {cortexxdr.shost}, initiated by user {cortexxdr.suser.0}.",
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json:3]: According to the suggestion in the parser:
```suggestion
    "value": "{cortexxdr.category}: An event of {cortexxdr.severity_human} severity was {event.action} on endpoint {host.name}, initiated by user {user.domain}\\{user.name}.",
```

File: Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    "conditions": [
+      {
+        "field": "cortexxdr.category"
+      },
+      {
+        "field": "cortexxdr.severity_human"
+      },
+      {
+        "field": "cortexxdr.action"
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json:12]:
```suggestion
        "field": "event.action"
```

File: Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      {
+        "field": "cortexxdr.shost"
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json:15]:
```suggestion
        "field": "host.name"
```

File: Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      {
+        "field": "cortexxdr.suser.0"
+      }
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json:19]:
```suggestion
      {
        "field": "user.domain"
      },
      {
        "field": "user.name"
      }
```

File: Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    ]
+  },
+  {
+    "value": "{cortexxdr.category}: {cortexxdr.original_message}",
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json:23]: The original message already exists in the field `message`
```suggestion
    "value": "{cortexxdr.category}: {message}",
```

File: Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    "conditions": [
+      {
+        "field": "cortexxdr.category"
+      },
+      {
+        "field": "cortexxdr.original_message"
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/smart-descriptions.json:29]:
```suggestion
```

File: Palo Alto Networks/paloalto-cortex/_meta/fields.yml [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  name: cortexxdr.shost
+  type: keyword
+cortexxdr.suser:
+  description: "User that created the event"
+  name: cortexxdr.suser
+  type: keyword
+cortexxdr.action:
+  description: "Action taken by cortexxdr"
+  name: cortexxdr.action
+  type: keyword
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/fields.yml:12]: According to the suggestion in the parser, these fields become unnecessary.
```suggestion
```

File: Palo Alto Networks/paloalto-cortex/_meta/fields.yml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  name: cortexxdr.event_subtype
+  type: keyword
+cortexxdr.severity_human:
+  description: "Severity of the event in human readable format. Can be UNKNOWN, LOW, MEDIUM or HIGH"
+  name: cortexxdr.severity_human
+  type: keyword
+cortexxdr.original_message:
+  description: "cortexxdr message parsed into json."
+  name: cortexxdr.original_message
+  type: keyword
```
Comment (squioc) [inline Palo Alto Networks/paloalto-cortex/_meta/fields.yml:28]: The field ECS [`message`](https://www.elastic.co/guide/en/ecs/8.4/ecs-base.html#field-message) already holds the original message.
```suggestion
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for our contribution.
Find below our suggestions for this format.

For any questions, don't hesitate to call our support or to see our [Contribution Guide](https://github.com/SEKOIA-IO/intake-formats/blob/main/CONTRIBUTING.md)

**Comment by m-242** [comment]
Thanks a lot for your review, I'll test these locally and come back to you.

**Comment by m-242** [comment]
I put @squioc 's recommandations in place, and fixed a few bugs and tests that came in the way.
One bug that I think is worth noting is the way the CEF `suser` field isn't parsed as a string and not an array, which forced me to use some buggy workaround in the parser that I explained in a comment.
This shouldn't be much of a problem for now, as cortex doesn't seem to send multiple users for events very often.

**Review by squioc — DISMISSED** [review]

**Comment by m-242** [comment]
I didn't mean to dismiss @squioc 's review, especially that I haven't changed anything in the PR except keeping the branch up to date with `main`.

**Review by squioc — APPROVED** [review]

**Comment by squioc** [comment]
@m-242 Don't worry. Github is very strict about merge and rebase.
To keep your branch updated, you can use the button 'Update branch'; Normally, that does not dismiss reviews.

**Comment by m-242** [comment]
Thanks a lot, I will in the future :)

**Pull request closed without merging**